test/instrumentation/modules/memcached.js

```diff
@@ -0,0 +1,52 @@
+'use strict'
+
+var agent = require('../../..').start({
+  serviceName: 'test',
+  secretToken: 'test',
+  captureExceptions: false,
+  metricsInterval: 0
+})
+
+var test = require('tape')
+var mockClient = require('../../_mock_http_client')
+test(function (t) {
+  resetAgent(function (data) {
+    t.end()
+  })
+  var Memcached = require('memcached')
+  var cache = new Memcached(`127.0.0.1:11211`, { timeout: 500 })
+  agent.startTransaction('foo', 'bar')
+  cache.set('foo', 'bar', 300, (err) => {
+    t.error(err)
+    cache.get('foo', (err, data) => {
+      t.error(err)
+      t.equal(data, 'bar')
+      cache.replace('foo', 'fizz', 300, (err) => {
+        t.error(err)
+        cache.get('foo', (err, data) => {
+          t.error(err)
+          t.equal(data, 'fizz')
+          cache.touch('foo', 300, (err) => {
+            t.error(err)
+            cache.del('foo', (err) => {
+              t.error(err)
+              cache.get('foo', (err, data) => {
+                t.error(err)
+                t.equal(data, undefined)
+                agent.endTransaction()
+                agent.flush()
+                t.end()
+              })
+            })
+          })
+        })
+      })
+    })
+  })
+})
+
+function resetAgent (cb) {
+  agent._instrumentation.currentTransaction = null
+  agent._transport = mockClient(11, cb)
```

Comment:
The callback to `resetAgent` is supposed to be called when the mock client has sent the number of events indicated here. In the above test, you only create 7 spans + 1 transaction - so 8 events in total - but it's waiting for 11. So the callback is never called. If you change it to 8, it will call the callback and you can do your final tests inside that.

```suggestion
  agent._transport = mockClient(8, cb)
```